WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModel.swift

```diff
@@ -51,6 +51,31 @@ final class PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewM
     }
 }
 
+extension PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModel: Hashable {
```

Comment:
The `hashValue` of a Hashable object should change when the (essential components of the) object change. For synthesised Hashable, that's whenever any property changes, and we should aim to get as close to that as possible.

> Hashing a value means feeding its essential components into a hash function, represented by the Hasher type. Essential components are those that contribute to the type’s implementation of Equatable.
> [...]
> In your hash(into:) implementation, call combine(_:) on the provided Hasher instance with the essential components of your type. To ensure that your type meets the semantic requirements of the Hashable and Equatable protocols, it’s a good idea to also customize your type’s Equatable conformance to match.
> – [Conforming to the Hashable protocol (Apple docs)](https://developer.apple.com/documentation/swift/hashable#Conforming-to-the-Hashable-Protocol)

Using (only) random IDs for hash values would lead to an incorrect implementation. 

Take the scenario where we make an object as follows:

```
struct Thing: Identifiable, Equatable {
    let id: String
    var title: String
}

extension Thing: Hashable {
    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}
```

Ignore that Hashable could be synthesised for this struct – we've written our own.

```
var foo = Thing(id: "123", title: "An awesome title")
let initialFoo = foo
let initialHash = foo.hashValue
foo.title = "Short title"
let laterHash = foo.hashValue
XCTAssertNotEqual(initialHash, laterHash) // Fails, because they are equal, but should pass with correct hashable implementation.
XCTAssertNotEqual(initialFoo, foo) // Passes, because equatable works fine
```

In SwiftUI, a change in the hash value gets used to determine whether the object has changed. Good objects to observe in SwiftUI will have a consistent ID (so we know it's the same object, and don't think the changed object is new) but a changing hashValue (so we know the object has updates and should be redrawn)

This comes in to play visually with animation and lists, but I think it also matters for deeper performance reasons in SwiftUI.

I'm going to double check my equatable implementations because I think there was one strange case where I was able to rely on the synthesised Equatable, but had to write a custom Hashable.